src/line/funcs/testAddValues.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @jest-environment jsdom
+ */
+const fs = require('fs');
+require('@testing-library/jest-dom');
+const domTesting = require('@testing-library/dom');
+const userEvent = require('@testing-library/user-event').default;
+const functions = require('./functions.js');
+
+const [htmlPath, jsPath] = functions.getPaths();
+
+// this function will perform the first test given in the "UI integration 
+// tests" section of the assignment document. The specific test is labeled as
+// "adding values in the chart builder".
+async function testAddValues(elements, inputs, num_empty_inputs){
+
+	// do initializations: load in files that we're gonna be testing, init
+	// expected length of (X, Y) pairs, grab the input fields, the '+' 
+	// button, and init our user for events
+    functions.initDomFromFiles(htmlPath, jsPath);
+	functions.initElements(elements);
+	const user = userEvent.setup();
+	var expected_length = 1;
```

Comment:
Based on how expected_length is used in this function, it should ideally be hard coded, so that the tester is explicitly expecting a specific number of values. Having expected_length being calculated is nice, but it should be slightly clearer and user defined than being automated, as all code can fail, including the code that is doing the test. So, making the test have as little automation as possible should be a goal.